packages/builder/src/utils/esbuild-index-html.ts

```diff
@@ -0,0 +1,60 @@
+import { ApplicationBuilderOptions } from "@angular-devkit/build-angular";
+import { Spinner } from "@angular/cli/src/utilities/spinner";
+import { env } from "@dotenv-run/core";
+import { DotenvRunOptions } from "@dotenv-run/webpack";
+import { readFile, writeFile } from "fs/promises";
+import { resolve } from "path";
+import { NgxEnvSchema } from "../builders/ngx-env/ngx-env-schema";
+import { variablesReducer } from "./variables-reducer";
+
+export async function indexHtml(
+  options: ApplicationBuilderOptions & NgxEnvSchema,
+  dotEnvOptions: DotenvRunOptions
+) {
+  const browserPath = resolve(
+    dotEnvOptions.cwd,
+    options.outputPath,
+    "browser/index.html"
+  );
+  const serverPath = resolve(
+    dotEnvOptions.cwd,
+    options.outputPath,
+    "server/index.server.html"
+  );
+  const raw = env(dotEnvOptions).raw;
+  const spinner = new Spinner();
```

Comment:
I think parsing the files should be pretty fast, there is no real need to add a spinner.
How did you manage to visualize it?

Reply:
for small projects it is definitely as you say, I worry about those projects where the css is inlined and they have a lot of scripts in the index

Reply:
let's keep it for fun